apps/software/lib/software/model/storage.ex

```diff
@@ -28,19 +29,6 @@ defmodule Helix.Software.Model.Storage do
   end
 
   @spec create_changeset() :: Ecto.Changeset.t
-  def create_changeset do
-    %__MODULE__{}
-    |> cast(%{}, [])
-    |> put_primary_key()
-  end
-
-  @spec put_primary_key(Ecto.Changeset.t) :: Ecto.Changeset.t
-  defp put_primary_key(changeset) do
-    if get_field(changeset, :storage_id) do
-      changeset
-    else
-      pk = PK.pk_for(__MODULE__)
-      put_change(changeset, :storage_id, pk)
-    end
-  end
+  def create_changeset,
+    do: cast(%__MODULE__{}, %{}, [])
```

Comment:
Also, how dialyzer behaves with default fields? It does also requires a typespec like 
```
@spec create_changeset(params) :: Ecto.Changeset.t
```
To work properly?